aten/src/ATen/native/native_functions.yaml

```diff
@@ -9351,6 +9351,18 @@
   dispatch:
     Math: linalg_tensorsolve_out
 
+- func: linalg_matrix_rank(Tensor self, float? tol=None, bool hermitian=False) -> Tensor
```

Comment:
This function should be marked as nondifferentiable in derivatives.yaml

Reply:
This function outputs a tensor of `torch.int64` dtype, so autograd can't work with it. Even if I add a nondifferentiable flag still it doesn't give a friendly message:
```python
In [7]: a = torch.randn(3, 10, 10, requires_grad=True)
In [8]: r = torch.matrix_rank(a)
In [9]: r
Out[9]: tensor([10, 10, 10]) # note no info on grad_fn
In [10]: r.backward()
RuntimeError: element 0 of tensors does not require grad and does not have a grad_fn
```